lib/index.js

```diff
@@ -84,21 +84,26 @@ const cli = co.wrap(function*(ar) {
     export: args.export,
     colors: args.colors,
     parsers: args.parsers, // to extend the default compiler parsers
-    from: args.from || args._.shift(),
-    to: args.to || args._.shift()
+    from: args.from || !args.stdin && args._.shift(),
+    to: args.to || !args.stdout && args._.shift(),
+    stdin: !!args.stdin,
+    stdout: !!args.stdout
   }
 
   // Call matching method
-  const method = Object
-                .keys(API)
-                .filter((v) => args[v])[0] || (opt.from ? 'make' : 'help')
+  const method =
+    Object.keys(API).filter(v => args[v])[0] ||
+    (opt.from || opt.stdin ? 'make' : 'help')
 
   // check whether the output should be colorized
   chalk.constructor({ enabled: !!opt.colors })
 
   // create isSilent as global variable
   global.isSilent = args.silent
 
+  // create isStdout as global variable
+  global.isStdout = args.stdout
```

Comment:
In the end we need to provide a log interface so we can abstract the logger in its own class/object and use  the `write()` method combined to the logger `type` property to log the cli messages